analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/SymbolicContextTest.cs

```diff
@@ -68,20 +69,20 @@ public void CreateSymbolicValue_UsesSymbolicValueCounter()
         }
 
         [TestMethod]
-        public void SetOperationConstraint_WithExistingValue_ReusesProgramState()
+        public void SetOperationConstraint_WithExistingValue()
         {
             var counter = new SymbolicValueCounter();
             var operation = CreateOperation();
             var state = ProgramState.Empty.SetOperationValue(operation, new SymbolicValue(counter));
 
             var sut = new SymbolicContext(counter, operation, state);
             var result = sut.SetOperationConstraint(DummyConstraint.Dummy);
-            result.Should().Be(state);
+            result.Should().NotBe(state, "new ProgramState instance should be created");
             result[operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue();
         }
 
         [TestMethod]
-        public void SetOperationConstraint_WithNewValue_CreatesNewProgramState()
+        public void SetOperationConstraint_WithNewValue()
```

Comment:
I like more the previous form which describes the expected behavior:
`SetOperationConstraint_WithNewValue_CreatesNewState`

Reply:
To me, that would be needed if any of those scenarios should do something different. In this case, the expected behavior is obvious from the assertions. Especially with the explicit message